test/e2e/tests/security-provider.spec.js

```diff
@@ -0,0 +1,255 @@
+              flagAsDangerous: 3,
+              reason:
+                'Because of an error, this request was not verified by the security. Proceed with caution.',
+              reason_header: 'Request not verified',
+            },
+          };
+        });
+        break;
+      default:
+        throw new Error(`Unknown scenario: ${scenario}`);
+    }
+  }
+
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+
+  it('Should return malicious response', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            transactionSecurityCheckEnabled: true,
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) =>
+          await mockSecurityProviderDetection(mockServer, 'malicious'),
+        dapp: true,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
+
+        testApp = windowHandles[1];
+
+        await driver.clickElement('#personalSign');
+
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        const warningHeader = await driver.isElementPresent({
+          text: 'This could be a scam',
+          tag: 'h5',
+        });
+        assert.equal(warningHeader, true);
+      },
+    );
+  });
+
+  it('Should return not safe response', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            transactionSecurityCheckEnabled: true,
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) =>
+          await mockSecurityProviderDetection(mockServer, 'notSafe'),
+        dapp: true,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
+
+        testApp = windowHandles[1];
```

Comment:
same here, I think we can remove this line as it's not needed anymore

Reply:
Done